pandas/core/algorithms.py

```diff
@@ -620,7 +620,10 @@ def rank(values, axis=0, method='average', na_option='keep',
         Whether or not to the display the returned rankings in integer form
         (e.g. 1, 2, 3) or in percentile form (e.g. 0.333..., 0.666..., 1).
     """
-    if values.ndim == 1:
+    if is_categorical(values):
```

Comment:
no, instead of here modify ``_get_data_algo``, check if its ``is_categorical_dtype``, then call ``values = values._values_for_rank()`` (we'll change the name below)